File: scratch/fix_css_indent.js
const fs = require('fs');
const path = require('path');

const filepath = path.join(__dirname, '..', 'assets', 'css', 'common.css');
const content = fs.readFileSync(filepath, 'utf8');
const lines = content.split('\n');

console.log('Total lines:', lines.length);

const newLines = [];
for (let i = 0; i < lines.length; i++) {
  const lineNum = i + 1; // 1-indexed
  let line = lines[i];

  if (lineNum >= 3675 && lineNum <= 5029) {
    // Remove leading 4 spaces of extra indentation
    if (line.startsWith('    ')) {
      line = line.substring(4);
    } else if (line.startsWith('  ') && line.trim().length > 0) {
      line = line.substring(2);
    }
    newLines.push(line);
  } else if (lineNum === 5030) {
    // This is the extra closing brace from the broken nesting - remove it
    const trimmed = line.trim();
    if (trimmed === '}') {
      console.log(`Removing extra closing brace at line ${lineNum}: '${line}'`);
      // Skip this line
      continue;
    } else {
      newLines.push(line);
    }
  } else {
    newLines.push(line);
  }
}

console.log('New total lines:', newLines.length);
fs.writeFileSync(filepath, newLines.join('\n'), 'utf8');
console.log('Done! File has been fixed.');

File: scratch/fix_css_indent.py
import os

filepath = r'd:\dgstore-main\dgstore-main\assets\css\common.css'

with open(filepath, 'r', encoding='utf-8') as f:
    lines = f.readlines()

print(f"Total lines: {len(lines)}")

# Lines 3675 to 5030 (0-indexed: 3674 to 5029) need un-indenting by 4 spaces
# Line 5030 (0-indexed: 5029) is the extra closing brace "  }" that needs to be removed

new_lines = []
for i, line in enumerate(lines):
    line_num = i + 1  # 1-indexed
    
    if 3675 <= line_num <= 5029:
        # Remove leading 4 spaces (or 2 levels of 2-space indent)
        if line.startswith('    '):
            line = line[4:]
        elif line.startswith('  '):
            line = line[2:]
        new_lines.append(line)
    elif line_num == 5030:
        # Skip this line - it's the extra closing brace from the broken nesting
        stripped = line.strip()
        if stripped == '}':
            print(f"Removing extra closing brace at line {line_num}: '{line.rstrip()}'")
            continue
        else:
            new_lines.append(line)
    else:
        new_lines.append(line)

print(f"New total lines: {len(new_lines)}")

with open(filepath, 'w', encoding='utf-8') as f:
    f.writelines(new_lines)

print("Done! File has been fixed.")
